alternative path for boost/uuid/sha1.hpp

Houkime opened the pull request:
Should fix https://github.com/anura-engine/anura/issues/283

Implementation of an answer here:
https://stackoverflow.com/a/33260104

Commit:
alternative path for boost/uuid/sha1.hpp

ghost commented:
Also note: https://github.com/ValveSoftware/steam-runtime/blame/efe918400a7b8294be66c2152f59ebfa2003cf97/README.md#L77-L82. For me that is a blocker _for raising the GCC version requirement_ unless someone else states otherwise.

hagabaka committed:
Merge branch 'trunk' into fix_boost

hagabaka committed:
Check for support of __has_include

This allows the code to be valid for compilers which do not suppot __has_include,
according to
https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005finclude.html

hagabaka committed:
Check of support of __has_include part 2

Add check for support of __has_include in formula_function.cpp
Check for the newer header instead of older one
(since if the boost is old, the compiler is more likely to be old too)
Reformat

hagabaka committed:
Fix boost uuid header missing else branch

Handle when the compiler supports __has_include,
but using old boost uuid header
